### PR TITLE
feat(asr): support speaker diarization

### PR DESCRIPTION
**Pull request by virajkarandikar**

### CODE REVIEW

File: riva/clients/asr/riva_asr_client.cc [hunk 1]
```diff
@@ -154,16 +156,27 @@ class RecognizeClient {
...
           auto& word_info = result.alternatives(0).words(w);
           std::cout << std::setw(40) << std::left << word_info.word();
-          std::cout << std::setw(16) << std::left << word_info.start_time();
-          std::cout << std::setw(16) << std::left << word_info.end_time() << std::endl;
+          if (word_time_offsets_) {
+            std::cout << std::setw(16) << std::left << word_info.start_time();
+            std::cout << std::setw(16) << std::left << word_info.end_time();
+          }
+          if (speaker_diarization_) {
+            std::cout << std::setw(16) << std::left << word_info.speaker_tag();
```
Comment (messiaen) [inline riva/clients/asr/riva_asr_client.cc]: Only write speaker tags for the first alternative.

**Review by messiaen — CHANGES_REQUESTED** [review]

File: riva/clients/asr/riva_asr_client.cc (hunk 1 above)
Comment (virajkarandikar) [reply riva/clients/asr/riva_asr_client.cc]: This loop runs only over words in first alternative.
`for (int w = 0; w < result.alternatives(0).words_size(); ++w) {`

Comment (messiaen) [reply riva/clients/asr/riva_asr_client.cc]: True.  Looks like you need to rebase.  Word timestamps and scores should be printed out for all alternatives.

**Review by messiaen — APPROVED** [review]

**Comment by messiaen** [comment]
@virajkarandikar Could you add the `--speaker_diarization` flag to the usage string.

I merge the diarization protos, so you can update WORKSPACE.

Except for the above I think this is good to go.

**Comment by virajkarandikar** [comment]
Sure. will update in couple of hours.

**Pull request merged**